File: CS/WpfApplication241/DateEditCalendarEx.cs
using DevExpress.Xpf.Editors.Popups.Calendar;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;
using System.Windows.Controls;

namespace WpfApplication241 {
    public class DateEditCalendarEx : DateEditCalendar {
        protected override void OnCellButtonClick(object sender, RoutedEventArgs e) {
            var button = sender as Button;
            if(button.IsEnabled) {
                base.OnCellButtonClick(sender, e);
            }

        }
    }
}

File: VB/WpfApplication241/DateEditCalendarEx.vb
Imports DevExpress.Xpf.Editors.Popups.Calendar
Imports System.Windows
Imports System.Windows.Controls

Namespace WpfApplication241

    Public Class DateEditCalendarEx
        Inherits DateEditCalendar

        Protected Overrides Sub OnCellButtonClick(ByVal sender As Object, ByVal e As RoutedEventArgs)
            Dim button = TryCast(sender, Button)
            If button.IsEnabled Then
                MyBase.OnCellButtonClick(sender, e)
            End If
        End Sub
    End Class
End Namespace
